stdlib/public/runtime/SwiftRT-WASM.cpp

```diff
@@ -0,0 +1,83 @@
+//===--- SwiftRT-WASM.cpp --------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "ImageInspectionCommon.h"
+#include "swift/shims/MetadataSections.h"
+
+#include <cstddef>
+#include <new>
+
+// Refer start/stop symbols weakly to link them if they aren't synthesized by the linker.
+#define DECLARE_SWIFT_SECTION(name)                                                             \
+  __attribute__((__visibility__("hidden"),__aligned__(1),weak)) extern const char __start_##name;    \
+  __attribute__((__visibility__("hidden"),__aligned__(1),weak)) extern const char __stop_##name;
+
+extern "C" {
+DECLARE_SWIFT_SECTION(swift5_protocols)
+DECLARE_SWIFT_SECTION(swift5_protocol_conformances)
+DECLARE_SWIFT_SECTION(swift5_type_metadata)
+
+DECLARE_SWIFT_SECTION(swift5_typeref)
+DECLARE_SWIFT_SECTION(swift5_reflstr)
+DECLARE_SWIFT_SECTION(swift5_fieldmd)
+DECLARE_SWIFT_SECTION(swift5_assocty)
+DECLARE_SWIFT_SECTION(swift5_replace)
+DECLARE_SWIFT_SECTION(swift5_replac2)
+DECLARE_SWIFT_SECTION(swift5_builtin)
+DECLARE_SWIFT_SECTION(swift5_capture)
+DECLARE_SWIFT_SECTION(swift5_mpenum)
+DECLARE_SWIFT_SECTION(swift5_accessible_functions)
+DECLARE_SWIFT_SECTION(swift5_runtime_attributes)
+}
+
+#undef DECLARE_SWIFT_SECTION
+
+namespace {
+static swift::MetadataSections sections{};
+}
+
+__attribute__((__constructor__))
+static void swift_image_constructor() {
+#define SWIFT_SECTION_RANGE(name)                                              \
+  { reinterpret_cast<uintptr_t>(&__start_##name),                              \
+    static_cast<uintptr_t>(&__stop_##name - &__start_##name) }
+
+  new (&sections) swift::MetadataSections {
+      swift::CurrentSectionMetadataVersion,
+      // NOTE: Multi images in a single process is not yet
+      // stabilized in WebAssembly toolchain outside of Emscripten.
+      { 0 },
```

Comment:
Totally appropriate to just leave the base address as `0` if it is meaningless. :)

Reply:
Thank you for clarification!

Reply:
Happy to help! You might want to consider using `nullptr` instead of `0` for clarity that this is a pointer, not an integer.

Reply:
I updated to use `nullptr` explicitly